app/src/main/java/com/space/quiz_app/presentation/utils/QuizUsernameValidation.kt

```diff
@@ -0,0 +1,25 @@
+package com.space.quiz_app.presentation.utils
+
+import androidx.annotation.StringRes
+import com.space.quiz_app.R
+
+//TODO New way to validate username
+enum class QuizUsernameValidation(@StringRes val errorText: Int)  {
+    USERNAME_INVALID_LENGTH_LONG(R.string.username_invalid_length_long),
+    USERNAME_INVALID_LENGTH_SHORT(R.string.username_invalid_length_short),
+    USERNAME_INVALID_CHARACTERS(R.string.username_invalid_characters),
+    LOGIN_SUCCESS(R.string.login_success);
+
+    companion object {
+        fun validate(input: String): QuizUsernameValidation {
+            return when {
+                input.length <= 8 -> USERNAME_INVALID_LENGTH_SHORT
+                input.length > 20 -> USERNAME_INVALID_LENGTH_LONG
+                !input.contains(QuizRegex.usernamePattern) -> USERNAME_INVALID_CHARACTERS
+                else -> LOGIN_SUCCESS
+            }
+        }
+    }
+
+}
+
```

Comment:
remove this class and move validations in usecase create error message strings in strings file